db/db_impl/db_impl.cc

```diff
@@ -1496,14 +1496,24 @@ ColumnFamilyHandle* DBImpl::PersistentStatsColumnFamily() const {
 Status DBImpl::Get(const ReadOptions& read_options,
                    ColumnFamilyHandle* column_family, const Slice& key,
                    PinnableSlice* value) {
+  return Get(read_options, column_family, key, value, nullptr);
+}
+
+Status DBImpl::Get(const ReadOptions& read_options,
+                   ColumnFamilyHandle* column_family, const Slice& key,
+                   PinnableSlice* value, std::string* timestamp) {
   GetImplOptions get_impl_options;
   get_impl_options.column_family = column_family;
   get_impl_options.value = value;
-  return GetImpl(read_options, key, get_impl_options);
+  Status s = GetImpl(read_options, key, get_impl_options);
+  if (timestamp && s.ok()) {
+    *timestamp = get_impl_options.timestamp;
```

Comment:
This is a second copy since get_impl_options.timestamp is of type `std::string`. Can we do better?

Reply:
Good point. Changed to pass in std::string* instead.